src/Kros.ApplicationInsights.Extensions/UserIdFromUserAgentInitializer.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Kros.ApplicationInsights.Extensions
+{
+    /// <summary>
+    /// Configuration of user id in Application Insights telemetry.
+    /// </summary>
+    /// <seealso cref="Microsoft.ApplicationInsights.Extensibility.ITelemetryInitializer" />
+    class UserIdFromUserAgentInitializer : ITelemetryInitializer
```

Comment:
Use explicit visibility modifier. Probabli `internal`.

Reply:
Write unit tests for this class.

Reply:
Added unit test and modifier